### PR TITLE
Resolve copypaste conflicts

### PR DESCRIPTION
**Pull request by 1ma**
When `xhgui-collector` was split from the main project `xhgui` was changed to depend on it, but the classes were copied rather than moved. This has led to 2 identical sets of files that must be kept in sync, and warnings in Composer since the class resolution is now ambiguous:

```bash
$ composer install --no-dev -a

Loading composer repositories with package information
Updating dependencies
Package operations: 5 installs, 0 updates, 0 removals
  - Installing slim/slim (2.6.3): Loading from cache
  - Installing slim/views (0.1.3): Loading from cache
  - Installing twig/twig (v1.35.2): Loading from cache
  - Installing pimple/pimple (v1.1.1): Loading from cache
  - Installing perftools/xhgui-collector (1.3.0): Loading from cache
Writing lock file
Generating optimized autoload files
Warning: Ambiguous class resolution, "Xhgui_Profiles" was found in both "$baseDir . '/src/Xhgui/Profiles.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Profiles.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Saver_Interface" was found in both "$baseDir . '/src/Xhgui/Saver/Interface.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Saver/Interface.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Saver_Mongo" was found in both "$baseDir . '/src/Xhgui/Saver/Mongo.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Saver/Mongo.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Saver_File" was found in both "$baseDir . '/src/Xhgui/Saver/File.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Saver/File.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Config" was found in both "$baseDir . '/src/Xhgui/Config.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Config.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Util" was found in both "$baseDir . '/src/Xhgui/Util.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Util.php", the first will be used.
Warning: Ambiguous class resolution, "Xhgui_Saver" was found in both "$baseDir . '/src/Xhgui/Saver.php" and "/home/marcel/workspace/xhgui/vendor/perftools/xhgui-collector/src/Xhgui/Saver.php", the first will be used.
```

A better approach is to distribute these classes among the two projects so there is no overlap. To that end I removed `Config`, `Util`, `Saver`, `Saver_File`, `Saver_Mongo` and `Saver_Interface` from the main project (since their content was still identical it is totally safe to do so). On the other hand, I removed `Profiles` from `xhgui-collector` and moved the little code it had into `Saver_Mongo`.

Doing this refactor a couple of Saver tests broke: they were being fed a string into their `save` methods instead of the expected array. These have been fixed.

`config.default.php` is not in sync in both projects, but ATM I'm not sure how to handle this, I'd like some input on that regard.

Note that `xhgui-collector` has been temporarily set to track my feature branch in order to demonstrate that the tests pass, but the collector's PR should be merged and released before merging this one (so I can undo the changes in the `composer.json` file).

### CODE REVIEW

File: tests/Saver/MongoTest.php [hunk 1]
```diff
@@ -4,16 +4,16 @@ class Saver_MongoTest extends PHPunit_Framework_TestCase
...
             ->disableOriginalConstructor()
             ->getMock();
-        $profiles->expects($this->once())
+        $collection->expects($this->once())
             ->method('insert')
-            ->with($this->equalTo($data));
+            ->with($this->equalTo($data + ['_id' => Xhgui_Saver_Mongo::getLastProfilingId()]));
 
-        $saver = new Xhgui_Saver_Mongo($profiles);
+        $saver = new Xhgui_Saver_Mongo($collection);
```
Comment (markstory) [inline tests/Saver/MongoTest.php]: Shouldn't this test move to the collector long term? Right now this is a test for a class from another package.

**Comment by markstory** [comment]
> config.default.php is not in sync in both projects, but ATM I'm not sure how to handle this, I'd like some input on that regard.

I'm not sure this needs to be addressed. I suspect overtime the 'viewer' app will have more configuration than the collector does. If we can build up tests in the collector to ensure it continues working as expected the config files don't need to be kept in sync.

File: tests/Saver/MongoTest.php (hunk 1 above)
Comment (1ma) [reply tests/Saver/MongoTest.php]: Definitely! However I don't have enough privileges to set up Travis for that project, so I didn't bother. But it should be done.

**Comment by 1ma** [comment]
Aite, then the `config.default.php` in the collector could be cleaned up a bit, it has a handful of settings that only apply to the viewer (`page.limit`, `templates.path`, etc).

**Comment by markstory** [comment]
I've done a release of the collector if you'd like to update this now.

**Comment by markstory** [comment]
Thanks @1m

**Pull request merged**